controller/program/src/instructions/admin/add_lst.rs

```diff
@@ -0,0 +1,161 @@
+        .with_pool_state(&POOL_STATE_ID)
+        .with_lst_state_list(&LST_STATE_LIST_ID)
+        .with_associated_token_program(&ATOKEN_ID)
+        .with_system_program(&SYS_PROG_ID)
+        .with_lst_token_program(token_prog)
+        .build();
+
+    verify_pks(abr, &accs.0, &expected_pks.0)?;
+    verify_signers(abr, &accs.0, &ADD_LST_IX_IS_SIGNER.0)
+        .map_err(|expected_signer| log_and_return_acc_privilege_err(abr, *expected_signer))?;
+
+    verify_tokenkeg_or_22_mint(lst_mint_acc)?;
+    verify_sol_value_calculator_is_program(abr.get(*accs.sol_value_calculator()))?;
+
+    // Verify no duplicate in lst state list
+    let list = LstStatePackedList::of_acc_data(abr.get(*accs.lst_state_list()).data())
+        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidLstStateListData))?;
+
+    if list.find_by_mint(lst_mint_acc.key()).is_some() {
+        return Err(Inf1CtlCustomProgErr(Inf1CtlErr::DuplicateLst).into());
+    }
+
+    verify_not_rebalancing_and_not_disabled(pool)?;
+
+    // Create pool reserves and protocol fee accumulator ATAs if they do not exist
+    if abr.get(*accs.pool_reserves()).data().is_empty() {
+        let create_accs = NewCreateIxAccsBuilder::start()
+            .with_funding(*accs.payer())
+            .with_ata(*accs.pool_reserves())
+            .with_wallet(*accs.pool_state())
+            .with_mint(*accs.lst_mint())
+            .with_sys_prog(*accs.system_program())
+            .with_token_prog(*accs.lst_token_program())
+            .build();
+
+        cpi.invoke_fwd(abr, &ATOKEN_ID, CreateIxData::as_buf(), create_accs.0)?;
+    }
+
+    if abr.get(*accs.protocol_fee_accumulator()).data().is_empty() {
+        let create_accs = NewCreateIxAccsBuilder::start()
+            .with_funding(*accs.payer())
+            .with_ata(*accs.protocol_fee_accumulator())
+            .with_wallet(*accs.protocol_fee_accumulator_auth())
+            .with_mint(*accs.lst_mint())
+            .with_sys_prog(*accs.system_program())
+            .with_token_prog(*accs.lst_token_program())
+            .build();
+
+        cpi.invoke_fwd(abr, &ATOKEN_ID, CreateIxData::as_buf(), create_accs.0)?;
+    }
+
+    // Realloc lst state list
+    let lst_state_list_acc = abr.get_mut(*accs.lst_state_list());
+    let old_acc_len = lst_state_list_acc.data_len();
+    lst_state_list_acc.grow_by(size_of::<LstStatePacked>(), false)?;
+
+    let new_acc_len = lst_state_list_acc.data_len();
+
+    let lamports_shortfall = Rent::get()?
+        .min_balance(new_acc_len)
+        .saturating_sub(Rent::get()?.min_balance(old_acc_len));
+
+    if lamports_shortfall > 0 {
+        cpi.invoke_fwd(
+            abr,
+            &SYS_PROG_ID,
+            TransferIxData::new(lamports_shortfall).as_buf(),
+            NewTransferIxAccsBuilder::start()
+                .with_from(*accs.payer())
+                .with_to(*accs.lst_state_list())
+                .build()
+                .0,
+        )?;
+    }
```

Comment:
`pay_for_rent_exempt_shortfall` looks like a useful util fn here, like what was done in flatslab. We're gonna need it again for anything that involves growing accounts (DisableList for eg). 

It can probably be improved upon from whats in flatslab:
- we dont need to pass account len as arg because we can just read it from the account and do it after the realloc has happened like what youve done here
- pass rent in as arg instead to allow users to only call `Rent::get()` in the outer scope once